### PR TITLE
Set Allowed Web Origins when creating client

### PR DESCRIPTION
**Pull request by andyhd**
Set Auth0 Allowed Web Origins on creating a client (Application)
* Comma-separated list of allowed origins for use with Cross-Origin Authentication and web message response mode
* This may remove spurious CORS error messages

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,6 +7,18 @@ WORKDIR /usr/src/app
 COPY requirements.txt ./
 RUN pip install --no-cache-dir -r requirements.txt
 
-COPY moj_analytics /usr/local/lib/python3.6/site-packages/moj_analytics
+COPY moj_analytics /usr/local/lib/python3.7/site-packages/moj_analytics
```
Comment (xoen) [inline Dockerfile:10]: If we're assuming Docker image has Python 3.7, wouldn't be better to use a more specific tag as a Base image?

(I find the [Python image tags](https://hub.docker.com/r/library/python/tags/) slightly confusing, would be `3.7-alpine` in this case?)

File: resource/out [hunk 2]
```diff
@@ -30,7 +30,8 @@ def create_client(src_dir, source={}, params={}):
     client = auth0.management.get_or_create(Client(
         name=source['app-name'],
         callbacks=[f'{app_url}/callback'],
-        allowed_origins=[app_url]))
+        allowed_origins=[app_url],
+        web_origins=[app_url]))
```
Comment (xoen) [inline resource/out]: I quickly looked for some Auth0 documentation regarding this, I didn't find much a part from a person who's [getting a `Additional properties not allowed: web_origins` error](https://community.auth0.com/t/setting-web-origins-when-creating-a-client-using-the-management-api/9130).

[Here](https://community.auth0.com/t/is-it-possible-to-add-allowed-web-origins-with-the-management-api/9385/1) someone suggest to create the client and then `PATCH` it but it's referring to API v2 🤷‍♂️ .

File: Dockerfile (hunk 1 above)
Comment (andyhd) [reply Dockerfile:10]: yes, i think this change was required due to the base image version being incremented. i will use a more specific base image tag.

File: resource/out (hunk 2 above)
Comment (andyhd) [reply resource/out]: interesting - i'll change it to be safe.

File: moj_analytics/auth0_client.py [hunk 3]
```diff
@@ -84,6 +84,16 @@ def create(self, resource):
 
         return resource.__class__(self, response)
 
+    def update(self, resource):
+        endpoint = '{}s'.format(resource.__class__.__name__.lower())
+
+        response = self.request('PATCH', endpoint, json=resource)
+
+        if 'error' in response:
+            raise UpdateResourceError(response.message)
```
Comment (xoen) [inline moj_analytics/auth0_client.py:97]: I didn't fire an editor so apologies if it's somewhere obvious: Where is `UpdateResourceError` imported/defined?

Also, what is the `if 'error' in response` expression does? I see we're using `requests`, isn't that raising an exception? Or it doesn't raise exceptions on `4xx`/`5xx`?

**Review by xoen — APPROVED** [review]
Looks good, I just asked a question on the error handling 👍

File: moj_analytics/auth0_client.py (hunk 3 above)
Comment (andyhd) [reply moj_analytics/auth0_client.py:97]: Good catch, I forgot to implement it.

Comment (andyhd) [reply moj_analytics/auth0_client.py:97]: The `response` is a JSON object, which may contain an `error` field. The error response does not raise an exception in Requests.

**Pull request merged**